### ISSUE TITLE
Add projects' interactive documentation

### ISSUE THREAD

**Issue by trallard** [Quansight-Labs/czi-scientific-python-mgmt#19]
## 📝 Summary

Expand the availability of interactive documentation examples and tutorials to at least five projects.

## 🚀 Tasks / Deliverables

TBD

## 📅 Estimated completion

[24 months milestone](https://github.com/Quansight-Labs/czi-scientific-python-mgmt/milestone/2)

## 📋 Additional information

-

**Comment by rgommers** [comment]
Year 1 goal of the first interactive docs for 1 project (SciPy) almost met, see gh-16.

@Carreau is now joining in to work on this larger year 2 goal.

**Comment by rgommers** [comment]
As a quick update:
- a lot of work was done on https://github.com/jupyterlite/jupyterlite-sphinx to make the user experience for interactive docs polished
- https://pywavelets.readthedocs.io/en/latest/ is deployed and contains buttons with each example in the reference guide to start JupyterLite and try things interactively
- the SciPy infra is all there, the switch just needs to be flipped now that SciPy 0.12.0 is becoming available in Pyodide 0.26.0 (not yet up on PyPI, but expected very soon).

**Comment by rgommers** [comment]
Also related: a fair amount of work is going into making binary wheels smaller (e.g. the machinery to strip off tests and data with a single build flag), reducing the number of build variants in compiled code, etc. Small binaries will be quite important to make the user experience good, because downloading many MBs of packages yields high latency and can be costly when a user doesn't have a connection with unlimited bandwidth.

**Comment by rgommers** [comment]
@agriyakhetarpal it may be useful to have a Status table in the issue description here (just like in pyodide/pyodide#18), linking to deployed interactive docs and/or open PRs. Would you be able to create one?

**Comment by agriyakhetarpal** [comment]
Yes, I'll do this in a moment, @rgommers. I expect that my interactive docs work will increase in magnitude in the coming months.

**Comment by Carreau** [comment]
With respect to installing nightly wheels when running jupyter-lite:

 - My own opinion is that one should install package when starting the kernel, cf `https://github.com/jupyterlite/pyodide-kernel/pull/158` (backticks for not crosslinking). I see installing those packages not as arbitrary code execution, as IMHO jupyter lite goal is to have a environment ready for end user to install.
 - There are some concern about load on Pypi and other indexes (despite CDN), and breaking pyodide itself.
 - As Agriya also pointed out there is `--piplite-wheel`, which allows embedding wheel into jupyter-lite, it raises the questions of dependencies (usually this is used for building a PR, but deps are pulled from indexes). My understanding is that even if we had `--piplite-wheel-and-all-deps`, they would still not allow install at kernel startup.
     - There is a current technical limitation that piplite/micropip cannot install only from the `--piplite-wheel`, but that's just a technical limitation.

Alternatives to a patch in pyodide-kernel are:
  - A fork. 
  - An Jupyterlite Extension
    - A full extension – not sure there are the right hooks ?
    - A shallow extension `pyodide-kernel-autoinstall` a shallow extension that just overrwite the startup method to install packages – but not sure how hard.
  - Auto inject a `await piplite.install([..packages..], index_urls=['proxy?'])` (potentially from a "file") to avoid a big cell. 
  - Patch in IPython if in emscripten to install things. 


See also `https://github.com/jupyterlite/jupyterlite/issues/508`, `https://github.com/jupyterlite/pyodide-kernel/issues/60`

**Comment by juntyr** [comment]
I've been working on a similar problem and am now using the following solution (not quite for micropip installs since I need locked environments):

- use micropip to create a custom lockfile that can be hosted elsewhere
- use my very bare-bones https://github.com/juntyr/a-jupyterlite-query-config extension to allow overriding the lockfile URL and preload package list via URL query parameters: https://github.com/climet-eu/lab/blob/8d33d03de19caf1a22f8611f4e634e69a6a1b931/jupyter-lite.json#L17
- now other projects can load their custom lock files and preload their packages without needing to deploy their own JupyterLite + Pyodide env

**Comment by rootsmusic** [comment]
@trallard Should [Scientific Python Lectures](https://lectures.scientific-python.org/) be added to your status table?

**Comment by agriyakhetarpal** [comment]
Hi @rootsmusic, I cannot answer for @trallard here of course, but I would believe that porting the Scientific Python Lectures to be interactive would be considered low-priority for now. Some of my thoughts are as follows:

- Those materials were set up with another Sphinx theme in mind, and swaying away from the PyData Sphinx Theme (which is one of the nicer ones in the avenue in terms of design, features, and accessibility) means that we need to create new styles for the jupyterlite-sphinx components to comply – unless they can switch themes. I don't immediately see a proposal to do so in their issue tracker.
- It's relatively easy to copy the notebooks and embed them in a JupyterLite gallery that's separate from the documentation. At the moment, you may even build/host your own site to do this with the notebooks downloaded from the repository. From a cursory look, I'm not entirely sure about how many notebooks will work seamlessly within a WASM environment and whether they have been updated to work with latest versions of Scientific Python libraries, but it could be worth a shot!
- Lastly, it would be nice if there were progress on scipy-lectures/scientific-python-lectures#781. It would help avoid redundant efforts around making them work well with interactive documentation and styling changes being discussed as a part of pyodide/pyodide#142. See also: scientific-python/summit-2025#5, where it would be nice to have consensus on the improvements to be made to these notes and their scope before putting Pyodide/JupyterLite forward as a pedagogical driver.
